lib/ohai/plugins/ec2.rb

```diff
@@ -51,9 +51,18 @@ def has_ec2_dmi?
   end
 
   # looks for a xen UUID that starts with ec2
-  # this gets us detection of Linux HVM and Paravirt hosts
+  # uses the sys tree on Linux and a WMI query on windows
+  # this gets us detection of HVM and Paravirt hosts
+  # @return [Boolean] do we have a Xen UUID or not?
   def has_ec2_xen_uuid?
-    if ::File.exist?("/sys/hypervisor/uuid")
+    if RUBY_PLATFORM =~ /mswin|mingw32|windows/
+      require "wmi-lite/wmi"
+      wmi = WmiLite::Wmi.new
+      if wmi.query("select uuid from Win32_ComputerSystemProduct")[0]["identifyingnumber"] =~ /^ec2/
```

Comment:
Isn't there a chance that `wmi.query(...)[0]` will return `nil` (in the event of an empty result set) and then you'd be trying to call `:[]` on `NilClass`?